docs: 📚 Document recurring problems in wrappers and web-components packages (#561)
synergy-design-system/synergy-design-system: packages/components/RECIPES.md

### Proposed Solution

Try to match on the next `<syn-nav-item>` with the closest selector. This will also match for slotted elements.

#### Solution:

```html
<!-- Before -->
<syn-nav-item href="/">
  Home
  <syn-icon name="home" slot="prefix"></syn-icon>
</syn-nav-item>

<script type="module">
  // This listener will only work when "Home" is clicked
  // It will display a runtime error (unknown property "href") when the syn-icon is clicked
  document.querySelector("syn-nav-item").addEventListener("click", e => {
    const { target } = e;
    document.location = target.ref;
  });
</script>

<!-- After -->
<syn-nav-item href="/">
  Home
  <syn-icon name="home" slot="prefix"></syn-icon>
</syn-nav-item>

<script type="module">

>   // This listener will only work when "Home" is clicked
>   // It will display a runtime error (unknown property "href") when the syn-icon is clicked

Reviewer: Wrong text (Copy paste)
```suggestion
  // This listener will work when "Home" or the syn-icon is clicked
```

Author: Fixed
